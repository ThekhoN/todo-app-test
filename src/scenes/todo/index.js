import React, { Component } from "react";
import TodoForm from "./components/todo-form";
import Header from "../../components/header";
import { connect } from "react-redux";
import { addTodo } from "../../redux/modules/todos";
import PropTypes from "prop-types";

class Todo extends Component {
	constructor() {
		super();
		this.state = {
			todo: ""
		};
	}
	render() {
		return (
			<div className="scene">
				<Header title="Todo App" />
				<br />
				<TodoForm onSubmit={this.props.addTodo} />
			</div>
		);
	}
}

Todo.propTypes = {
	addTodo: PropTypes.func.isRequired
};

const mapStateToProps = state => ({ todos: state.todos.todos });

const mapDispatchToProps = dispatch => ({
	addTodo: payload => dispatch(addTodo(payload))
});

export default connect(mapStateToProps, mapDispatchToProps)(Todo);
