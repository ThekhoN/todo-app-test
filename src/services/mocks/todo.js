export const todoData = [
  {
    id: 1,
    text: "Implement React Tests",
    status: "doing"
  },
  {
    id: 2,
    text: "Cook",
    status: "to do"
  }
];

export const todoObj = {
  id: 1,
  text: "Implement React Tests",
  status: "doing"
};
