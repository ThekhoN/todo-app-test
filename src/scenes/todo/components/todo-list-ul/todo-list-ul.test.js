import React from "react";
import TodoList from "./index";
import { shallow, mount } from "enzyme";
import { todoData as data } from "../../../../services/mocks/todo";
import TodoListLi from "../todo-list-li";

describe("TodoList", () => {
  let wrapper, todoData;
  beforeEach(() => {
    todoData = data;
    wrapper = shallow(<TodoList todoData={todoData} />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains number of TodoListLi equal to todoData.length", () => {
    expect(wrapper.find(TodoListLi).length).toBe(todoData.length);
  });
});
