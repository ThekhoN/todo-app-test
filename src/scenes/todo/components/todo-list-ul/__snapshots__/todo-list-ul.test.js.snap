// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoListUl
    todoData={
      Array [
        Object {
          "id": 1,
          "status": "doing",
          "text": "Implement React Tests",
        },
        Object {
          "id": 2,
          "status": "to do",
          "text": "Cook",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TodoListLi
          todoObj={
            Object {
              "id": 1,
              "status": "doing",
              "text": "Implement React Tests",
            }
          }
        />,
        <TodoListLi
          todoObj={
            Object {
              "id": 2,
              "status": "to do",
              "text": "Cook",
            }
          }
        />,
      ],
      "className": "todo-list-ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "todoObj": Object {
            "id": 1,
            "status": "doing",
            "text": "Implement React Tests",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "todoObj": Object {
            "id": 2,
            "status": "to do",
            "text": "Cook",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TodoListLi
            todoObj={
              Object {
                "id": 1,
                "status": "doing",
                "text": "Implement React Tests",
              }
            }
          />,
          <TodoListLi
            todoObj={
              Object {
                "id": 2,
                "status": "to do",
                "text": "Cook",
              }
            }
          />,
        ],
        "className": "todo-list-ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "todoObj": Object {
              "id": 1,
              "status": "doing",
              "text": "Implement React Tests",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "todoObj": Object {
              "id": 2,
              "status": "to do",
              "text": "Cook",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
